src/mlpack/core/data/dictionary_encoding_impl.hpp

```diff
@@ -0,0 +1,166 @@
+/**
+ * @file dictionary_encoding_impl.hpp
+ * @author Jeffin Sam
+ *
+ * Implementation of dictionary encoding functions.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_DICTIONARY_ENCODING_IMPL_HPP
+#define MLPACK_CORE_DATA_DICTIONARY_ENCODING_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "dictionary_encoding.hpp"
+
+namespace mlpack {
+namespace data {
+
+void DictionaryEncoding::Reset()
+{
+  mappings.clear();
+  originalStrings.clear();
+}
+
+DictionaryEncoding::DictionaryEncoding(const DictionaryEncoding& oldObject) :
+    originalStrings(oldObject.originalStrings)
+{
+  std::deque<std::string>::iterator jt = originalStrings.begin();
+  for (auto it = oldObject.originalStrings.begin(); it !=
+      oldObject.originalStrings.end(); it++)
+  {
+    mappings[*jt] = oldObject.mappings.at(*it);
+    jt++;
+  }
+}
+
+DictionaryEncoding DictionaryEncoding::operator= (const
+    DictionaryEncoding &oldObject)
+{
+  if (this != &oldObject)
+  {
+    mappings.clear();
+    originalStrings.clear();
+    originalStrings = oldObject.originalStrings;
+    std::deque<std::string>::iterator jt = originalStrings.begin();
+    for (auto it = oldObject.originalStrings.begin(); it !=
+        oldObject.originalStrings.end(); it++)
+    {
+      mappings[*jt] = oldObject.mappings.at(*it);
+      jt++;
+    }
+  }
```

Comment:
Looks like you forgot to return the object.

Reply:
I thought while calling `obj1 = obj2`, it internally changes to `obj1.operator=(obj2)` and thus it manipulating the same object
https://github.com/jeffin143/mlpack/blob/7a31397fece32cf2a39fd2b5404f63be845839c9/src/mlpack/core/data/dictionary_encoding_impl.hpp#L39-L47

Reply:
Correct, but you should be able to write something like `a = b = c;`. Hence you have to return the object itself.